kobo/apps/superuser_stats/views.py

```diff
@@ -22,40 +31,227 @@ def _base_filename_to_full_filename(base_filename, username):
 
 @user_passes_test(lambda u: u.is_superuser)
 def country_report(request):
+    """
+    Generates a report which counts the number of submissions and forms
+    per country as reported by the user
+    """
+    today = timezone.now().date()
     base_filename = 'country-report_{}_{}_{}.csv'.format(
         re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
-        date.today(),
+        today,
         datetime.now().microsecond
     )
 
-    # Get the date date filters from the query and set defaults
-    start_date = request.GET.get('start_date', date.today())
-    tomorrow = date.today() + timedelta(days=1)
+    # Get the date filters from the query and set defaults
+    start_date = request.GET.get('start_date', today)
+    tomorrow = today + timedelta(days=1)
     end_date = request.GET.get('end_date', tomorrow)
 
     # Generate the CSV file
     filename = _base_filename_to_full_filename(
         base_filename, request.user.username)
     generate_country_report.delay(filename, start_date, end_date)
```

Comment:
This also doesn't seem to be working as all the countries in the report show 0 counts. Maybe we should also only include counties that have at least one count?

Reply:
There is definitely something with this report with dev server. Let me check

Reply:
Still seems like there are some issues:
- The export still contains a long list of countries with 0 counts
- Only one country has a count above 0

Seems there are a few data types that this needs to handle which is a bit messy:
```python
>>> pprint(list(User.objects.all().values('extra_details__data__country')))
[{'extra_details__data__country': 'CAN'},
 {'extra_details__data__country': None},
 {'extra_details__data__country': ''},
 {'extra_details__data__country': None},
 {'extra_details__data__country': 'NPL'},
 {'extra_details__data__country': 'MWI'},
 {'extra_details__data__country': None},
 {'extra_details__data__country': None},
 {'extra_details__data__country': ''},
 {'extra_details__data__country': 'POL'},
 {'extra_details__data__country': [{'label': 'Canada', 'value': 'CAN'},
                                   {'label': 'South Africa', 'value': 'ZAF'}]},
 {'extra_details__data__country': None},
 {'extra_details__data__country': [{'label': 'Albania', 'value': 'ALB'}]},
 {'extra_details__data__country': ''},
 {'extra_details__data__country': ''},
 {'extra_details__data__country': None}]
```